src/Render/SvgFlatRender.php

```diff
@@ -11,130 +11,14 @@
-    <text x="{{ vendorStartPosition }}" y="15" fill="#010101" fill-opacity=".3">{{ vendor }}</text>
-        <text x="{{ vendorStartPosition }}" y="14">{{ vendor }}</text>
-        <text x="{{ valueStartPosition }}" y="15" fill="#010101" fill-opacity=".3">{{ value }}</text>
-        <text x="{{ valueStartPosition }}" y="14">{{ value }}</text>
-    </g>
-</svg>
-EOF;
-*/
-
-private static $template  = <<<EOF
-<svg xmlns="http://www.w3.org/2000/svg" width="{{ totalWidth }}" height="20">
-    <linearGradient id="b" x2="0" y2="100%">
-    <stop offset="0" stop-color="#bbb" stop-opacity=".1"/>
-    <stop offset="1" stop-opacity=".1"/>
-    </linearGradient>
-    <mask id="a">
-    <rect width="{{ totalWidth }}" height="20" rx="3" fill="#fff"/>
-    </mask>
-    <g mask="url(#a)">
-    <rect width="{{ vendorWidth }}" height="20" fill="#555"/>
-    <rect x="{{ vendorWidth }}" width="{{ valueWidth }}" height="20" fill="{{ valueColor }}"/>
-    <rect width="{{ totalWidth }}" height="20" fill="url(#b)"/>
-    </g>
-    <g fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="11">
-    <text x="{{ vendorStartPosition }}" y="15" fill="#010101" fill-opacity=".3">{{ vendor }}</text>
-    <text x="{{ vendorStartPosition }}" y="14">{{ vendor }}</text>
-    <text x="{{ valueStartPosition }}" y="15" fill="#010101" fill-opacity=".3">{{ value }}</text>
-    <text x="{{ valueStartPosition }}" y="14">{{ value }}</text>
-    </g>
-</svg>
-EOF;
-
-    /**
-     * Constructor.
-     *
-     * @param TextSizeCalculatorInterface $textSizeCalculator
-     */
-    public function __construct(TextSizeCalculatorInterface $textSizeCalculator = null)
-    {
-        $this->textSizeCalculator = $textSizeCalculator;
-
-        if (null === $this->textSizeCalculator) {
-            $this->textSizeCalculator = new GDTextSizeCalculator();
-        }
-    }
-
-    /**
-     * @param Badge $badge
-     *
-     * @return mixed
-     */
-    public function render(Badge $badge)
-    {
-        $parameters = array();
-
-        $parameters['vendorWidth']         = $this->stringWidth($badge->getSubject());
-        $parameters['valueWidth']          = $this->stringWidth($badge->getStatus());
-        $parameters['totalWidth']          = $parameters['valueWidth'] + $parameters['vendorWidth'];
-        $parameters['vendorColor']         = self::VENDOR_COLOR;
-        $parameters['valueColor']          = $badge->getHexColor();
-        $parameters['vendor']              = $badge->getSubject();
-        $parameters['value']               = $badge->getStatus();
-        $parameters['vendorStartPosition'] = round($parameters['vendorWidth'] / 2, 1) + 1;
-        $parameters['valueStartPosition']  = $parameters['vendorWidth'] + round($parameters['valueWidth'] / 2, 1) - 1;
-
-        return $this->renderSvg(self::$template, $parameters, $badge->getFormat());
-    }
+    protected $templateName = 'flat';
```

Comment:
expecting child class to overwrite a protected property to initialize it looks weird to me